### ISSUE TITLE
Unable to download artifacts larger than 2^32-1 bytes

### ISSUE THREAD

**Issue by gbhand** [dawidd6/action-download-artifact#109]
Really appreciate the work you've done here!

I've recently been trying to download a 3GB artifact and keep running into this error:
![download-artifact-error (2)](https://user-images.githubusercontent.com/8967600/135152616-3b8ea28b-742c-446d-84a2-d0fcd120dfd6.png)

I'm fairly certain this is a file size issue as uploading a file exactly 2147483646 bytes in size works fine:
![download-artifact-success](https://user-images.githubusercontent.com/8967600/135152223-344a4590-b0ab-40cb-bda4-b05bbd5e6e74.png)

I believe this is the line that throws the error https://github.com/dawidd6/action-download-artifact/blob/af92a8455a59214b7b932932f2662fdefbd78126/main.js#L138
It seems like the Buffer objects can only hold 2^32-1 bytes.

I'd make a fork myself (maybe reading the zip's data as a stream?) but I'm pretty novice with NodeJS. Handling larger files could be a useful option but I understand the project may have other plans.

Thanks!

**Comment by dawidd6** [comment]
I had trouble here in the past making it work in JavaScript. Would appreciate a PR making things better in terms of handling bigger files.

**Comment by radimkohout** [comment]
In my head, there is one possible solution, which is to download artifact to file, unZIP it, end then delete the ZIP file. The downside is, that you need to have double the space. If this is an acceptable solution, I can make it, and PR it.

**Comment by dawidd6** [comment]
Would be great if you or someone else could first confirm this workaround really fixes the issue. If it does, then we can introduce a new switch.

**Comment by radimkohout** [comment]
Ok, give me some time.

**Comment by radimkohout** [comment]
I can't find any proper way, how to do it in node, it could be done by command line tools (wget on Linux, Invoke-WebRequest/ something else in Windows PowerShell).

**Comment by daleyjem** [comment]
Can't you just add an action input to not unpack, and let the consumer do that themselves if they want to (using `tar` or something)?

**Comment by radimkohout** [comment]
@daleyjem I've done something like that there: https://github.com/GarnetOS/action-download-artifact
**But note, that it's Linux dependent, and needs wget and unzip installed.**

**Comment by daleyjem** [comment]
Thanks @radimkohout. That's not quite what I meant though. I just want to be able to download an unpacked/unzipped artifact. Once I get it, I can just add my own extra `step` to unpack it, or do whatever I want with it. For me, the real "gold" with this action is being able to download from another `run`

I submitted a pull request with what I'm talking about: https://github.com/dawidd6/action-download-artifact/pull/150

**Comment by radimkohout** [comment]
@daleyjem Yes, I know, but it is my solution for my usage, where I need large artifacts. So I know, it is not ideal, but it is usable currently.

**Comment by dawidd6** [comment]
Switch for skipping the unpacking process has been added.
